e2e/test/iothub/properties/PropertiesWithComponentsE2ETests.cs

```diff
@@ -472,7 +472,7 @@ private async Task Properties_DeviceSendsNullValueForPropertyResultsServiceRemov
 
             // For the property patch sent here will result in the component being removed.
             var propertyPatch4 = new ClientPropertyCollection();
-            propertyPatch4.AddComponentProperties(ComponentName, null);
+            propertyPatch4.AddRootProperty(ComponentName, null);
```

Comment:
This is an oddity. Since we don't have the plural overload anymore, we can't null out the entire component collection. Instead we need to treat it like a root property.

@abhipsaMisra, thoughts?

Reply:
Implementation wise, this should be fine. Any property with a null value will end up getting removed. This does feel a bit weird naming wise, though.

Reply:
I feel kinda weird about `AddRootProperty` instead of `SetRootProperty`. Thoughts on the verb?